### PR TITLE
fix(operaor): handle apparmor and security fs mounts

### PR DESCRIPTION
**Pull request by rksharma95**
**Purpose of PR?**:
This PR handles the cases:
- `/etc/apparmor.d` is not present on host to be mounted with `KubeArmor daemonset` when apparmor enforcer is being used.
- `/sys/kernel/security` is not present to be mounted with `kubearmor-controller`. 

when the operator is used to deploy the KubeArmor. In both the cases, if required path is not present, kubearmor darmonset or controller will be deployed without the mount. 

Fixes #

**Does this PR introduce a breaking change?**
No
**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Comment by daemon1024** [comment]
@VedRatan to test it out.

**Comment by VedRatan** [comment]
Policies are not behaving as expected but the kubearmor-operator is running. @daemon1024 @rksharma95 .

**Comment by rksharma95** [comment]
@VedRatan can you paste the karmor probe result here thanks

**Comment by VedRatan** [comment]
image

@rksharma95  for your reference.

**Review by VedRatan — APPROVED** [review]
lgtm

**Comment by daemon1024** [comment]
Why is kubearmor not running? Audit Policies should work?

**Comment by daemon1024** [comment]
You should see details about kubearmor in karmor probe

File: pkg/KubeArmorOperator/enforcer/enforcer.go [hunk 1]
```diff
@@ -27,7 +27,25 @@ func CheckBtfSupport(PathPrefix string, log zap.SugaredLogger) string {
 	return "no"
 }
 
-// DetectEnforcer: detect the enforcer on the node
+// CheckIfApparmorFsPresent checks if BTF is present
+func CheckIfApparmorFsPresent(PathPrefix string, log zap.SugaredLogger) string {
+	btfPath := PathPrefix + "/etc/apparmor.d"
+	if _, err := os.Stat(filepath.Clean(btfPath)); err == nil {
```
Comment (nyrahul) [inline pkg/KubeArmorOperator/enforcer/enforcer.go]:
```suggestion
	path := PathPrefix + "/etc/apparmor.d"
	if _, err := os.Stat(filepath.Clean(path)); err == nil {
```

File: pkg/KubeArmorOperator/enforcer/enforcer.go [hunk 2]
```diff
@@ -27,7 +27,25 @@ func CheckBtfSupport(PathPrefix string, log zap.SugaredLogger) string {
...
+		return "yes"
+	}
+	return "no"
+}
+
+// CheckIfSecurityFsPresent checks if Security filesystem is present
+func CheckIfSecurityFsPresent(PathPrefix string, log zap.SugaredLogger) string {
+	btfPath := PathPrefix + "/sys/kernel/security"
+	if _, err := os.Stat(filepath.Clean(btfPath)); err == nil {
```
Comment (nyrahul) [inline pkg/KubeArmorOperator/enforcer/enforcer.go]:
```suggestion
	path := PathPrefix + "/sys/kernel/security"
	if _, err := os.Stat(filepath.Clean(path)); err == nil {
```

**Review by daemon1024 — APPROVED** [review]
Verified the changes, LGTM

**Pull request merged**